### ISSUE TITLE
enh: prevent closing "New Message" window with ESC key and clicking outside the modal box

### ISSUE THREAD

**Issue by funkycram** [humhub/humhub-modules-mail#167]
### To reproduce the problem
Write a new message (a modal box is shown), and click outside the modal box.
If you have already written a message, you loose all your text !

### Enhancement
Don't close the modal box if :
- you click outside the modal box
- you press the ESC key on the keyboard

**Comment by funkycram** [comment]
I share my custom script I had to put in the theme to solve this problem, replacing the file `view/mail/create.php` and adding this script at the end of the file, and `id="create-message-form"` in the first `div` of this view :

```
<script type="text/javascript">
    // Prevent closing modal box while writing a message
    // Add `id="create-message-form"` in the first `div` of this view
    $('#create-message-form button[class="close"], #create-message-form button[data-dismiss="modal"]').on('click', function(e) {
        if (
            $('textarea#newMessageText').val() != ''
            && !confirm('Attention, ton message en cours de rédaction n‘est pas sauvé ! Est-tu sûr(e) de vouloir fermer cette fenêtre ?')
        ) {
            e.stopPropagation();
            return false;                
        }
        else {
            needToConfirm = false;
            return true;
        }
    });
    $('#globalModal').prepend('<div style="width:100%; height: 100%; position: absolute;"></div>');
</script>
```

**Comment by funkycram** [comment]
Solved with Humhub 1.6
